fluent-dom/src/localization.js

```diff
@@ -55,17 +55,49 @@ export default class Localization {
    */
   async formatWithFallback(keys, method) {
     const translations = [];
+    let hasMissing = true;
+    let debugInfo = null;
+
     for (let ctx of this.ctxs) {
       // This can operate on synchronous and asynchronous
       // contexts coming from the iterator.
       if (typeof ctx.then === "function") {
         ctx = await ctx;
       }
-      const errors = keysFromContext(method, ctx, keys, translations);
-      if (!errors) {
+      if (keysFromContext(method, ctx, keys, translations)) {
+        hasMissing = false;
         break;
+      } else {
+        if (debugInfo === null) {
+          debugInfo = new Map();
+        }
+        debugInfo.set(
+          ctx.locales[0],
+          keys
+            .filter((key, i) => translations[i] === undefined)
+            .map(key => key[0])
+        );
+      }
+    }
+
+    if (hasMissing) {
+      keys.forEach((key, i) => {
+        if (translations[i] === undefined) {
+          translations[i] = { value: key[0], attrs: null };
+        }
+      });
+    }
+    if (debugInfo !== null && typeof console !== "undefined") {
+      if (hasMissing) {
+        console.warn("[Fluent] Could not resolve all keys.");
+      } else {
+        console.warn("[Fluent] Had to use fallback to resolve all keys.");
+      }
+      for (const [locale, ids] of debugInfo.entries()) {
+        console.warn(`Tried "${locale}". Missing keys - ${ids.join(", ")}`);
```

Comment:
This is the most important information here. I think the minimal desired output should be something like:

    Missing translations in es-MX: foo, bar.
    Missing translations in es: baz.
    Missing translations in en-US: qux.

IIUC, you could return the list of missing keys from `keysFromContext` and print them right in the `for (let ctx of this.ctxs)` loop.